qutebrowser/misc/tabsfinder.py

```diff
@@ -0,0 +1,82 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+from collections import OrderedDict, namedtuple
+
+from qutebrowser.utils import objreg
+
+Tab = namedtuple('Tab', 'url title index')
+
+
+class TabsFinder:
+    """
+    TabsFinder is in charge of gathering a reusable list of
+    tabs amongst all windows
+
+    Args:
+        tabbed_browser:
+        skip_win_id: The id of the window to skip, or None to include all.
+    """
+
+    def __init__(self, tabbed_browser=None, skip_win_id=None):
+        self._tabbed_browser = tabbed_browser
+        self._skip_win_id = skip_win_id
+        self._tabs = []
+        self._tabs_by_win_id = OrderedDict()
+
+        self._find()
+
+    def _find(self):
+        """Find all the tabs
+
+        """
+
+        for win_id in objreg.window_registry:
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=win_id)
+            self._tabs_by_win_id[win_id] = []
+
+            if self._skip_win_id is not None and win_id == self._skip_win_id:
+                continue
+
+            if tabbed_browser.shutting_down:
+                continue
+
+            for idx in range(tabbed_browser.widget.count()):
+                tab = tabbed_browser.widget.widget(idx)
+                t = Tab(tab.url().toDisplayString(),
+                        tabbed_browser.widget.page_title(idx),
+                        "{}/{}".format(win_id, idx + 1))
+
+                self._tabs.append(t)
+                self._tabs_by_win_id[win_id].append(t)
+
+    def tabs_by_window(self):
+        """
+        Returns an iteratable set of items from win_id to tabs
+        """
+
+        return self._tabs_by_win_id.items()
+
+    def tabs(self):
+        return self._tabs
+
+    def delete_tab(self, data):
+        """Close the selected tab."""
+
+        if self._tabbed_browser is None:
```

Comment:
`self._tabbed_browser` is only really used here. I don't think it makes sense to pass it in the constructor to the class, but rather to this function directly (so it can't be called without specifying a tabbed browser).

Also, I don't think this function needs to strictly be a class method, it might be fine with just being a static method or not in this class at all, so we can call it without needing to run `_find`.

Reply:
Fixed.